ResearchKit/Common/ORKMedicationPicker.h

```diff
@@ -0,0 +1,28 @@
+//
+//  ORKMedicationPicker.h
+//  ResearchKit
+//
+//  Created by Eric Schramm on 5/30/18.
+//  Copyright © 2018 researchkit.org. All rights reserved.
+//
+
+@import UIKit;
+
+@class ORKMedicationPicker;
+@class ORKMedication;
+
+NS_ASSUME_NONNULL_BEGIN
+
+@protocol ORKMedicationPickerDelegate <NSObject>
+
+- (void)medicationPicker:(ORKMedicationPicker *)medicationPicker didSelectMedication:(ORKMedication *)medication;
+
+@end
+
+@interface ORKMedicationPicker : UITableViewController
```

Comment:
I get what you're doing here, but I think I'd rather see this as a protocol than the sort-of abstract class you're providing here. You could implement DummyMedPicker as a UITableViewController that calls out to the delegate as an example, but if ORKMedicationPicker was a protocol the implementer could use a UICollectionViewController or their own UI paradigm to implement the picker. (Even something as simple as a UIPickerView could work in some situations.)